tests/test_operator.py

```diff
@@ -950,6 +953,37 @@ def test_incomplete_override(self):
 
 class TestDeclarator(object):
 
+    def test_conditional_declarations(self):
```

Comment:
this is test is much more complicated than strictly necessary

why do you need all that ListInitializer stuff?

also, if we have this test, can we not drop the one in OPS? they are identical...

Reply:
Just explaining a little more, this test is to ensure that the case below doesn't occur (undeclared array inside else)
```
if (x < 3)
{
  int s2d_name_4pt[8] = {0, 0, 0, 1, 1, 0, 1, 1};
}
else
{
  s2d_name_4pt[8] = {0, 0, 0, 1, 1, 0, 1, 1};
}
```

> why do you need all that ListInitializer stuff?

When the array isn't initialized an **AssertionError** occurs during `ccode` generation.
```
devito/ir/iet/visitors.py:259: in visit_Conditional
    then_body = c.Block(self._visit(o.then_body))
self = <cgen.Block object at 0x7f6dcb2c9450>, contents = (None,)
for item in contents:
   assert isinstance(item, Generable)
```


> also, if we have this test, can we not drop the one in OPS? they are identical...

They are identical! The reason is that the backend OPS overwrites Array in devito.ops.types.Array. It is possible that future changes in ...ops.Array may break something from this fix. Once `test_operator.py` has skipif(['yask', 'ops']), backend ops stay uncovered.

Reply:
I still don't think that 30 almost identical lines are duplicated across two different files

>  It is possible that future changes in ...ops.Array may break something from this fix

if you further change ops.Array, then that's when you should write a new test, I feel

Reply:
nitpicking: this is an advanced test, so it should be put at the bottom of the `TestDeclarator` class, not at the top

Reply:
> I still don't think that 30 almost identical lines are duplicated across two different files

I agree!

> if you further change ops.Array, then that's when you should write a new test, I feel

I sincerely believe that testing should come before code evolution to ensure project integrity, but as I agree with the previous point and I don't have a solution for code duplicity, so I removed it.